pkg/ingester/ingester.go

```diff
@@ -2079,10 +2080,35 @@ func wrappedTSDBIngestErr(ingestErr error, timestamp model.Time, labels []mimirp
 		return nil
 	}
 
-	return fmt.Errorf(errTSDBIngest, ingestErr, timestamp.Time().UTC().Format(time.RFC3339Nano), mimirpb.FromLabelAdaptersToLabels(labels).String())
+	var errID globalerror.ID
+	var errMsg string
+	switch cause := errors.Cause(ingestErr); cause {
+	case storage.ErrOutOfBounds:
+		errID = globalerror.OutOfBounds
+		errMsg = "something out of bounds"
+	case storage.ErrOutOfOrderSample:
+		errID = globalerror.OutOfOrderSample
+		errMsg = "something sample out of order"
+	case storage.ErrDuplicateSampleForTimestamp:
+		errID = globalerror.DuplicateSampleForTimestamp
+		errMsg = "something duplicate sample for timestamp"
```

Comment:
Error message could be something like "the sample has been rejected because another sample with the same timestamp but a different value has already been ingested"